paderbox/utils/misc.py

```diff
@@ -74,3 +53,67 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         sys.stdout.close()
         sys.stdout = self._original_stdout
+
+
+def all_equal(x: Iterable[Hashable]) -> bool:
+    """
+    Checks if all elements in `x` are equal. Returns `False` if `x` is empty.
+
+    Defined to improve readability.
+
+    Examples:
+        >>> all_equal((1, 1, 1, 1))
+        True
+        >>> all_equal((1, 1, 1, 2))
+        False
+    """
+    return len(set(x)) == 1
+
+
+def all_unique(x: Iterable[Hashable]) -> bool:
+    """
+    Checks if all elements in `x` are unique. Returns `True` if `x` is empty.
+
+    Defined to improve readability.
+
+    Examples:
+        >>> all_unique((1, 2, 3, 4))
+        True
+        >>> all_unique((1, 2, 3, 1))
+        False
+    """
+    return len(set(x)) == len(list(x))
+
+
+def all_in(x: Iterable[Hashable], y: Iterable[Hashable]) -> bool:
+    """
+    Check if all elements in `x` are in `y`. Returns `True` if `x` is empty.
+
+    Equivalent to `set(x).issubset(y)`, but faster.
```

Comment:
This depends. I expected that `set(x).issubset(y)` is in relevant cases faster (i.e. large x and large y and return is True).

Reply:
I checked again and, at least for integers and strings, `set(s).issubset(y)` is always faster. I don't know what I did when I wrote the function and the comment. I'll change it to `set(s).issubset(y)`.

Reply:
You use a generator comprehension in combination with `all`.
When one of the first entries is False, this is usually faster (i.e. expected output is `False`).
Maybe you had such a case in the past.